Meta/CMake/libweb_generators.cmake

```diff
@@ -97,94 +97,62 @@ function (generate_js_bindings target)
-                global-mixin-header
-                global-mixin-implementation
-            )
         endif()
 
+        list(TRANSFORM BINDINGS_SOURCES PREPEND "${CMAKE_CURRENT_BINARY_DIR}/")
         target_sources(${target} PRIVATE ${BINDINGS_SOURCES})
-        # FIXME: cmake_minimum_required(3.17) for ZIP_LISTS
-        list(LENGTH BINDINGS_SOURCES num_bindings)
-        math(EXPR bindings_end "${num_bindings} - 1")
-        foreach(iter RANGE "${bindings_end}")
-            list(GET BINDINGS_SOURCES ${iter} bindings_src)
-            list(GET BINDINGS_TYPES ${iter} bindings_type)
-            get_property(include_paths DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
-            list(TRANSFORM include_paths PREPEND -i)
-
-            # Ninja expects the target name in depfiles to be relative to CMAKE_BINARY_DIR, but ${bindings_src} is
-            # relative to CMAKE_CURRENT_BINARY_DIR. CMake >= 3.20 can do the rewriting transparently (CMP0116).
-            if(CMAKE_GENERATOR MATCHES "^Ninja" AND NOT POLICY CMP0116)
-                # FIXME: Drop this branch for cmake_minimum_required(3.20)
-                get_filename_component(full_path ${bindings_src} ABSOLUTE BASE_DIR ${CMAKE_CURRENT_BINARY_DIR})
-                file(RELATIVE_PATH depfile_target ${CMAKE_BINARY_DIR} ${full_path})
-            else()
-                set(depfile_target ${bindings_src})
-            endif()
-
-            add_custom_command(
-                OUTPUT "${bindings_src}"
-                COMMAND "$<TARGET_FILE:Lagom::BindingsGenerator>" "--${bindings_type}" -o "${bindings_src}.tmp" --depfile "${bindings_src}.d"
-                        --depfile-target "${depfile_target}" ${include_paths} "${LIBWEB_INPUT_FOLDER}/${class}.idl" "${LIBWEB_INPUT_FOLDER}"
-                COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${bindings_src}.tmp" "${bindings_src}"
-                COMMAND "${CMAKE_COMMAND}" -E remove "${bindings_src}.tmp"
-                VERBATIM
-                DEPENDS Lagom::BindingsGenerator
-                MAIN_DEPENDENCY ${class}.idl
-                DEPFILE ${CMAKE_CURRENT_BINARY_DIR}/${bindings_src}.d
-            )
-        endforeach()
-
-        foreach(generated_file IN LISTS BINDINGS_SOURCES)
-            get_filename_component(generated_name ${generated_file} NAME)
-            add_custom_target(generate_${generated_name} DEPENDS ${generated_file})
-            add_dependencies(all_generated generate_${generated_name})
-            add_dependencies(${target} generate_${generated_name})
-        endforeach()
+
+        get_property(include_paths DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+        list(TRANSFORM include_paths PREPEND -i)
+
+        # Ninja expects the target name in depfiles to be relative to CMAKE_BINARY_DIR, but ${bindings_src} is
+        # relative to CMAKE_CURRENT_BINARY_DIR. CMake >= 3.20 can do the rewriting transparently (CMP0116).
```

Comment:
Maybe I am missing here, but looks like the `FIXME` comment is being dropped here for `cmake_minimum_required(3.20)`, and our root `CMakeLists` has a minimum required for 3.25. Should we perhaps be enabling that policy?

Reply:
Meta/Lagom/CMakeLists.txt has a minimum required of 3.16 still

Reply:
Ah thanks! Sorry for the noise! Thid looks like a nice simplification 🙂

Reply:
Yep no worries. We enable the policy at the top of Meta/CMake/common_options.cmake